mypyc/ops.py

```diff
@@ -322,9 +322,12 @@ def __hash__(self) -> int:
 class Environment:
     """Maintain the register symbol table and manage temp generation"""
 
-    def __init__(self) -> None:
+    def __init__(self, name: Optional[str] = None) -> None:
+        self.name = name
         self.indexes = OrderedDict()  # type: Dict[Value, int]
-        self.symtable = {}  # type: Dict[Var, Register]
+        self.symtable = {}  # type: Dict[SymbolNode, Register]
+        self.functions = {}  # type: Dict[str, FuncDef]
+        self.ret_type = none_rprimitive  # type: RType
```

Comment:
I don't think that `Environment` is the right place to track the current return type.
Probably the right thing to do is to maintain a stack of `ret_types`, like is done for `environments` and a few other things.

I kind of don't like that either, though. I think the long-term right thing to do is to pull all of the the things that `IRBuilder` keeps stacks of into one class, and keep of stack of that. No need to do that in this PR, though.